### PR TITLE
Update for Swift 2.2

### PR DESCRIPTION
**Pull request by jorystiefel**
Replaces C-style `for` loops, `var` in function declarations, and `i++` increment operators.

### CODE REVIEW

File: Heimdall/Heimdall.swift [hunk 1]
```diff
@@ -882,18 +894,21 @@ private extension NSData {
                 offset += OID.count
                 
                 // Type
-                if bytes[offset++] != 0x03 {
+                if bytes[offset] != 0x03 {
```
Comment (henrinormak) [inline Heimdall/Heimdall.swift:897]: Should the contents of this if also increment the `offset` key? If it is true, then currently the offset value is not incremented, which did happen previously (right?).

File: Heimdall/Heimdall.swift [hunk 2]
```diff
@@ -882,18 +894,21 @@ private extension NSData {
...
                 }
                 
+                offset += 1
+                
                 // Skip over the contents length field
                 let _ = NSInteger(octetBytes: bytes, startIdx: &offset)
                 
                 // Contents should be separated by a null from the header
-                if bytes[offset++] != 0x00 {
+                if bytes[offset] != 0x00 {
```
Comment (henrinormak) [inline Heimdall/Heimdall.swift:907]: Same comment here about offset not being incremented if this if is true.

**Comment by henrinormak** [comment]
Added some comments, otherwise good to go.

File: Heimdall/Heimdall.swift (hunk 1 above)
Comment (jorystiefel) [inline Heimdall/Heimdall.swift:897]: This should be correct. If this is true, then the `if` statement does `return self` and there is no point in incrementing the `offset`, as the entire function exits.

File: Heimdall/Heimdall.swift (hunk 2 above)
Comment (jorystiefel) [inline Heimdall/Heimdall.swift:907]: Same response as above :)

**Comment by henrinormak** [comment]
Everything seems in order, merging. I will push a new release to CocoaPods shortly after.

**Comment by jorystiefel** [comment]
Thanks Henri!

**Pull request merged**